components/main/SearchModal.tsx

```diff
@@ -21,16 +22,16 @@ const SearchModal = React.memo(function SearchModal(props: ModalProps) {
   useEffect(() => {
     clear();
     return () => {};
-  }, [props.value]);
+  }, [searchModal]);
 
   const filteredData = SEARCH_SUGGESTIONS_DUMMY.filter((e) =>
     search === ""
       ? true
       : e.title.toLowerCase().includes(search.trim().toLowerCase()),
   );
   return (
-    <ModalTemplate {...props} title="Search Tuturku" fullscreen>
-      <div className="flex flex-col gap-2 sm:gap-4">
+    <ModalTemplate value={searchModal} onClose={closeSearchModal} title="Search Tuturku" fullscreen>
```

Comment:
Might need to change this thing, because currently i got some z-index issue on the bottom bar when every time i switched from the search tab, the BottomBar flicker a lil bit, because the z-index of ModalTemplate is set higher than the BottomBar, fixing this thing rn.

Reply:
i think we can leave this for now at least. And move on to the more funcitonality side of things.

Reply:
It's okay i think i will be done soon enough